### PR TITLE
objc: remove deprecated APIs

### PR DESCRIPTION
**Pull request by hajimehoshi**
Perhaps, do we need new examples?

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Updated with RegisterClass.

File: objc/objc_runtime_darwin.go [hunk 1]
```diff
@@ -583,7 +554,7 @@ type IMP uintptr
 func NewIMP(fn interface{}) IMP {
 	ty := reflect.TypeOf(fn)
 	if ty.Kind() != reflect.Func {
-		panic("objc: not a function")
+		panic(fmt.Sprintf("objc: not a function"))
```
Comment (TotallyGamerJet) [inline objc/objc_runtime_darwin.go:557]: Why the change to Sprintf?

**Review by TotallyGamerJet — APPROVED** [review]
Other than comment LGTM

File: objc/objc_runtime_darwin.go (hunk 1 above)
Comment (hajimehoshi) [reply objc/objc_runtime_darwin.go:557]: oops this was my mistake

**Comment by TotallyGamerJet** [comment]
LGTM
